provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitionerrole/PractitionerRoleTransformer.java

```diff
@@ -1,3 +1,94 @@
+import gov.va.api.health.providerdirectory.service.PpmsPractitionerRole;
+import gov.va.api.health.providerdirectory.service.PpmsProviderSpecialtiesResponse;
+import gov.va.api.health.providerdirectory.service.ProviderContacts;
+import gov.va.api.health.providerdirectory.service.ProviderResponse;
+import gov.va.api.health.providerdirectory.service.controller.EnumSearcher;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PractitionerRoleTransformer implements PractitionerRoleController.Transformer {
+
+  @Value("${provider-directory.url")
+  String baseUrl;
+
+  @Override
+  public PractitionerRole apply(PpmsPractitionerRole ppms) {
+    // TODO organization reference is required
+    // location could be populated by caresites
+    ProviderResponse.Value provider = ppms.providerResponse().value().get(0);
+    return PractitionerRole.builder()
+        .resourceType("PractitionerRole")
+        .active(StringUtils.equalsIgnoreCase(provider.providerStatusReason(), "active"))
+        .practitioner(practitionerReference(provider))
+        .code(
+            CodeableConcept.builder()
+                .coding(codeCodings(ppms.providerSpecialtiesResponse()))
+                .build())
+        .telecom(
+            ppms.providerContacts().value().isEmpty()
+                ? telecoms(null)
+                : ppms.providerContacts()
+                    .value()
+                    .stream()
+                    .flatMap(v -> telecoms(v).stream())
+                    .collect(Collectors.toList()))
+        .build();
+  }
+
+  private List<Coding> codeCodings(PpmsProviderSpecialtiesResponse specialtiesResponse) {
+    return specialtiesResponse
+        .value()
+        .stream()
+        .map(v -> Coding.builder().code(v.codedSpecialty()).build())
+        .collect(Collectors.toList());
+  }
+
+  private Reference practitionerReference(ProviderResponse.Value provider) {
+    return Reference.builder()
+        .reference(baseUrl + "/Practitioner/" + provider.providerIdentifier())
+        .build();
+  }
+
+  PractitionerContactPoint telecom(String system, String value) {
+    return PractitionerContactPoint.builder()
+        .system(EnumSearcher.of(ContactPoint.ContactPointSystem.class).find(system))
+        .value(value)
+        .build();
+  }
+
+  List<PractitionerContactPoint> telecoms(ProviderContacts.Value source) {
+    if (source == null
+        || allBlank(source.mobilePhone(), source.businessPhone(), source.email(), source.fax())) {
+      return null;
+    }
+    List<PractitionerContactPoint> telecoms = new ArrayList<>();
+    if (source.mobilePhone() != null) {
+      telecoms.add(telecom("phone", source.mobilePhone()));
```

Comment:
Change this code slightly to let the `telecom` method handle the null/blank checks:

```
PractitionerContactPoint phone = telecom("phone", source.mobilePhone());
    if (phone != null) {
      telecoms.add(phone);
    }
```